tests/components/blink/test_diagnostics.py

```diff
@@ -0,0 +1,34 @@
+"""Test AccuWeather diagnostics."""
+from unittest.mock import MagicMock, patch
+
+from syrupy import SnapshotAssertion
+from syrupy.filters import props
+
+from homeassistant.core import HomeAssistant
+
+from tests.components.diagnostics import get_diagnostics_for_config_entry
+from tests.typing import ClientSessionGenerator
+
+YAML_CONFIG = {"username": "test-user", "password": "test-password"}
+
+
+async def test_entry_diagnostics(
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+    snapshot: SnapshotAssertion,
+    mock_blink_api: MagicMock,
+    mock_config_entry: MagicMock,
+) -> None:
+    """Test config entry diagnostics."""
+
+    mock_config_entry.add_to_hass(hass)
+
+    with patch("homeassistant.components.blink.Blink", return_value=mock_blink_api):
```

Comment:
Since you are already adding fixture it might be nicer to make one that takes care of patching this with the default mock